File: pe-core/src/main/java/org/saar/core/common/r2d/Model2D.java
package org.saar.core.common.r2d;

import org.saar.core.node.Model;

public class Model2D implements Model, Node2D {

    private final Mesh2D mesh;

    public Model2D(Mesh2D mesh) {
        this.mesh = mesh;
    }

    @Override
    public void draw() {
        this.mesh.draw();
    }

    @Override
    public void delete() {
        this.mesh.delete();
    }
}

File: pe-core/src/main/java/org/saar/core/common/r2d/Model2D.kt
package org.saar.core.common.r2d;

import org.saar.core.mesh.Model;

public class Model2D implements Model {

    private final Mesh2D mesh;

    public Model2D(Mesh2D mesh) {
        this.mesh = mesh;
    }

    @Override
    public Mesh2D getMesh() {
        return this.mesh;
    }
}
